File: Part1_강의자료/Part1_연습문제_풀이/10986_나머지합/10986.cpp
#include <stdio.h>
#include <vector>
using namespace std;

int main()
{
    int N, M;
    scanf("%d %d", &N, &M);

    vector<int> vec(N + 1);
    for (int i = 1; i <= N; i++)
    {
        int num;
        scanf("%d", &num);
        vec[i] = num % M;
    }

    vector<int> acc_remain_cnt(M);
    int cur_remain = 0;
    acc_remain_cnt[cur_remain]++;
    long long ans = 0;

    for (int i = 1; i <= N; i++)
    {
        cur_remain = (cur_remain + vec[i]) % M;
        ans += acc_remain_cnt[cur_remain];
        acc_remain_cnt[cur_remain]++;
    }
    printf("%lld\n", ans);
    return 0;
}

File: Part1_강의자료/Part1_연습문제_풀이/10986_나머지합/10986.py
N, M = map(int, input().split())
li = list(map(int, input().split()))

acc_remain_cnt = [0] * M
acc_remain_cnt[cur_remain := 0] = 1
ans = 0

for x in li:
    ans += acc_remain_cnt[cur_remain := (cur_remain + x) % M]
    acc_remain_cnt[cur_remain] += 1

print(ans)
